### DISCUSSION TITLE
Python Linting

### DISCUSSION THREAD

**Discussion by krishnagogada** [ajaxorg/ace#4840] [Q&A] [answered]
We want linting for python. We tried using [skulpt](https://github.com/bnmnetp/skulpt)(https://github.com/ajaxorg/ace/pull/1908). 
What we planned is from [skulpt](https://github.com/bnmnetp/skulpt) we get error type, error message, and line number so that using this data we can set annotation using `setAnnotations` method. But what happens is skulpt complies the python code and gives only one output.

From below image output shows only 'a' is not defined but here 'b' is also not defined.
![image](https://user-images.githubusercontent.com/54623105/178724820-cd9d495c-3c42-4b2f-b52f-c4f03e581c43.png)

Is there any other way for implementing python linting?

**Comment by andrewnester** [comment, marked as answer]
Following this page you can right your own ace worker which will do validation for python using your suggested library

https://github.com/ajaxorg/ace/wiki/Syntax-validation
